rpc/chain/client.go

```diff
@@ -0,0 +1,691 @@
+package chain
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+
+	"github.com/afex/hystrix-go/hystrix"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/status-im/status-go/services/rpcstats"
+)
+
+type FeeHistory struct {
+	BaseFeePerGas []string `json:"baseFeePerGas"`
+}
+
+type ClientWithFallback struct {
+	ChainID  uint64
+	main     *ethclient.Client
+	fallback *ethclient.Client
+
+	mainRpc     *rpc.Client
+	fallbackRpc *rpc.Client
+
+	IsConnected bool
+}
+
+func NewSimpleClient(main *rpc.Client, chainID uint64) *ClientWithFallback {
+	hystrix.ConfigureCommand(fmt.Sprintf("ethClient_%d", chainID), hystrix.CommandConfig{
+		Timeout:               10000,
+		MaxConcurrentRequests: 100,
+		SleepWindow:           300000,
+		ErrorPercentThreshold: 25,
+	})
+
+	return &ClientWithFallback{
+		ChainID:     chainID,
+		main:        ethclient.NewClient(main),
+		fallback:    nil,
+		mainRpc:     main,
+		fallbackRpc: main,
```

Comment:
Doesn't this mean it will retry two times? I'm wondering if this is intentional and if there is no problem retrying using the same client.